### PR TITLE
Feature/book details, update and delete

### PR DESCRIPTION
**Pull request by DanaBrasanac1901**

### CODE REVIEW

File: src/components/bookDelete/BookDelete.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { useNavigate, useParams } from 'react-router-dom'
+
+import { DialogContentProps } from '../../interfaces/DialogContentProps'
+import { deleteBook } from '../../services/BookService'
+
+export function BookDelete({ submitClickEvent, isModalFirstRender, setIsModalReadyToClose }: DialogContentProps ){
+  const bookId = useParams().bookId
+  const navigate = useNavigate()
+  useEffect(() => {
+    if(isModalFirstRender || !bookId) return
```
Comment (aleksandarkrnjaic) [inline src/components/bookDelete/BookDelete.tsx:12]: What is the purpose of this prop? (isModalFirstRender)

Comment (DanaBrasanac1901) [reply src/components/bookDelete/BookDelete.tsx:12]: when modal gets opened,   useEffect for form gets triggered and it sends an api request, so isModalFirstRender is used to check  if the submit button was actually clicked or the modal just opened

File: src/services/BookService.tsx [hunk 2]
```diff
@@ -13,3 +14,18 @@ export const getAllBooksPaginated = async ({ pageNumber, pageLength }: Paginated
   if(url) return axios.get(`${url}Books/paged`, { params: { pageNumber: pageNumber, pageLength: pageLength } } )
   return Promise.reject(Error('URL not valid'))
 }
+
+export const getBookDetails = async (bookId : number) : Promise<AxiosResponse<BookDetailsResponse>>=> {
+  if(url) return axios.get(`${url}Books/${bookId}`)
```
Comment (SaleBojke) [inline src/services/BookService.tsx]: uvek ce ti pokupiti iz env file-a url, ako ga ne pokupi bice bad request, tako da ti je nepotrebna provera if(url)

Comment (DanaBrasanac1901) [reply src/services/BookService.tsx]: To mi je marko rekao da valja proveriti

**Review by SaleBojke — APPROVED** [review]

**Pull request merged**